김진/231219/다트게임.js

```diff
@@ -0,0 +1,36 @@
+function solution(dartResult) {
+    let scoreArr = [];
+    let resultArr = [];
+    let num = '' ;
+    
+    for (let i = 0; i < dartResult.length; i++) { //배열로 나누기 위한 for문
+      if (!isNaN(dartResult[i])) {
+        num += dartResult[i];
+      } else {
+        scoreArr.push(num);
+        scoreArr.push(dartResult[i]);
+        num = '';
+      }
+    }
+    scoreArr = scoreArr.filter(Boolean); //공백 없이 배열을 반환
+    
+    for (let i = 0; i < scoreArr.length; i++) {
+      if (scoreArr[i] === 'S') { //S 이면 1제곱
+        resultArr.push(scoreArr[i-1]);
+      }
+      else if (scoreArr[i] === 'D') {
+        resultArr.push(scoreArr[i-1] * scoreArr[i-1]); //D 이면 2제곱
+      }
+      else if (scoreArr[i] === 'T') {
+        resultArr.push(scoreArr[i-1] * scoreArr[i-1] * scoreArr[i-1]); //T 이면 3제곱
```

Comment:
```suggestion
        resultArr.push(scoreArr[i-1] ** 2); //D 이면 2제곱
      }
      else if (scoreArr[i] === 'T') {
        resultArr.push(scoreArr[i-1] ** 3); //T 이면 3제곱
```
제곱의 경우에 **를 통해 표현하면 코드가 좀 더 간결해질 것 같아요
https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Operators/Exponentiation